buildzaltodistdir

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -e
+
+params=$@
+if [ "$#" == 1 ]; then
+  if [ $1 == "all" ]; then
+    params=(8.6.0 8.7.0 8.7.1 8.7.2 8.7.3 8.7.4 8.7.5 8.7.6 8.7.7 8.7.9 8.7.10 8.7.11 8.8.2 8.8.3 8.8.4 8.8.5 8.8.6 8.8.7 8.8.8 8.8.9 8.8.10 8.8.11 8.8.12 8.8.15 9.0.0)
+  fi
+fi
+
+N=4
+i=0
+for j in ${params[@]}; do
+  ((i=$i+1)); (($i%$N==0)) && wait
+   mvn install -Pcompatibility-check-output-dir -Dzimbra.version=$j &
+done
+wait
```

Comment:
I've passed `shellcheck` and used `parallel` to run a parallel number of mvn invocations equal to half the number of processors one has, and removed the workarounds with `wait`. Now shellchecks doens't complain anymore and the script works like a charm, Ctrl+C handling included :fire: 
```suggestion
params=( "$@" )
concurrent_jobs="$(($(nproc)/2))"

echo "Performing ${concurrent_jobs} jobs at the same time"

if [ "${#}" == 1 ]; then
    if [ "${1}" == "all" ]; then
        params=(8.6.0 8.7.0 8.7.1 8.7.2 8.7.3 8.7.4 8.7.5 8.7.6 8.7.7 8.7.9 8.7.10 8.7.11 8.8.2 8.8.3 8.8.4 8.8.5 8.8.6 8.8.7 8.8.8 8.8.9 8.8.10 8.8.11 8.8.12 8.8.15 9.0.0)
    fi
fi

printf '%s\0' "${params[@]}" | parallel -kj"${concurrent_jobs}" -0 "mvn install -Pcompatibility-check-output-dir -Dzimbra.version={} || echo 'error while processing Openzal for Zimbra {}'; exit 1"

```

The output is printed according the input provided, i.e if the input is `8.6.0 8.7.7 8.8.15` the resulting output would be respectively for `8.6.0` and so on, while still executing in parallel. To grab an error you just have to execute the script and do `| grep 'error while processing'` :wink: